File: skills/cow-arms-technology-analysis/scripts/R/05_descriptive_table.R
#!/usr/bin/env Rscript
# 05_descriptive_table.R — first-adoption year of assault rifle (s7), by region.
#
# Outputs a journal-ready descriptive table (CSV + LaTeX). Demonstrates the
# leap-frog check from codebook §3.1: a country whose s7_use jumps 0 → 9 with
# no intervening 1 adopted a superior tech without ever using the assault rifle.
# Here s7 is the most advanced small arm, so leap-frog cannot occur — but the
# check pattern generalizes.

suppressPackageStartupMessages({
  library(readr)
  library(dplyr)
  library(tidyr)
})

data_dir <- Sys.getenv("COW_ARMS_DATA_DIR",
                       unset = file.path("..", "..", "..", "..", "ArmsTechnologyV1"))

long <- read_csv(file.path(data_dir, "cow_arms_tech_long.csv"),
                 show_col_types = FALSE) |>
  filter(!is.na(techname))

# Minimal region recoding — illustration only. For a real paper, merge with an
# established regional classification (e.g. UN M.49, COW region codes).
region_for <- function(ccode) {
  case_when(
    ccode >= 2   & ccode <= 199 ~ "Americas",
    ccode >= 200 & ccode <= 399 ~ "Europe",
    ccode >= 400 & ccode <= 599 ~ "Sub-Saharan Africa",
    ccode >= 600 & ccode <= 699 ~ "MENA",
    ccode >= 700 & ccode <= 999 ~ "Asia–Pacific",
    TRUE                        ~ NA_character_
  )
}

first_adopt <- long |>
  filter(techname == "Assault rifle", use == 1) |>
  group_by(ccode, statename) |>
  summarise(first_year = min(year), .groups = "drop") |>
  mutate(region = region_for(ccode))

# Region-level summary
tbl <- first_adopt |>
  group_by(region) |>
  summarise(n_countries = n(),
            earliest = min(first_year),
            median = median(first_year),
            latest = max(first_year),
            .groups = "drop") |>
  arrange(median)

print(tbl)

dir.create("output", showWarnings = FALSE)
write_csv(tbl, "output/table_first_adoption_assault_rifle.csv")

# LaTeX output (replace the line below with kableExtra::kable() for full styling)
sink("output/table_first_adoption_assault_rifle.tex")
cat("\\begin{tabular}{lrrrr}\n\\hline\n")
cat("Region & N countries & Earliest & Median & Latest \\\\\n\\hline\n")
for (i in seq_len(nrow(tbl))) {
  row <- tbl[i, ]
  cat(sprintf("%s & %d & %d & %d & %d \\\\\n",
              row$region, row$n_countries, row$earliest, row$median, row$latest))
}
cat("\\hline\n\\end{tabular}\n")
sink()
cat("\nWrote output/table_first_adoption_assault_rifle.{csv,tex}\n")

File: skills/cow-arms-technology-analysis/scripts/python/05_descriptive_table.py
#!/usr/bin/env python3
"""05_descriptive_table.py — first-adoption year of the assault rifle, by region.

Outputs CSV + LaTeX. Demonstrates a region-summary pattern. For a real paper,
merge against an established regional classification (UN M.49 or COW regions)
rather than the illustrative ccode-range bins below.
"""

import os
from pathlib import Path
import pandas as pd

data_dir = Path(os.environ.get(
    "COW_ARMS_DATA_DIR",
    Path(__file__).resolve().parents[3] / "ArmsTechnologyV1",
))

long = pd.read_csv(data_dir / "cow_arms_tech_long.csv")
long = long[long["techname"].notna()]   # drop the total_use rows

# Illustrative region bins by ccode range. Replace with a proper classification
# for actual research.
def region_for(ccode: int) -> str:
    if   2   <= ccode <= 199: return "Americas"
    elif 200 <= ccode <= 399: return "Europe"
    elif 400 <= ccode <= 599: return "Sub-Saharan Africa"
    elif 600 <= ccode <= 699: return "MENA"
    elif 700 <= ccode <= 999: return "Asia–Pacific"
    return "Unknown"

first_adopt = (
    long.loc[(long.techname == "Assault rifle") & (long.use == 1)]
    .groupby(["ccode", "statename"])["year"].min()
    .reset_index(name="first_year")
)
first_adopt["region"] = first_adopt["ccode"].map(region_for)

tbl = (
    first_adopt.groupby("region")["first_year"]
    .agg(n_countries="size", earliest="min", median="median", latest="max")
    .reset_index()
    .sort_values("median")
)
tbl["median"] = tbl["median"].astype(int)
print(tbl.to_string(index=False))

Path("output").mkdir(exist_ok=True)
tbl.to_csv("output/table_first_adoption_assault_rifle.csv", index=False)

# LaTeX
with open("output/table_first_adoption_assault_rifle.tex", "w") as f:
    f.write("\\begin{tabular}{lrrrr}\n\\hline\n")
    f.write("Region & N countries & Earliest & Median & Latest \\\\\n\\hline\n")
    for _, r in tbl.iterrows():
        f.write(f"{r.region} & {r.n_countries} & {r.earliest} & "
                f"{r['median']} & {r.latest} \\\\\n")
    f.write("\\hline\n\\end{tabular}\n")
print("\nWrote output/table_first_adoption_assault_rifle.{csv,tex}")
